ui/pages/confirmations/components/confirm/header/header-info.tsx

```diff
@@ -200,7 +204,7 @@ const HeaderInfo = ({
             <ConfirmInfoRow label="Balance">
               <ConfirmInfoRowCurrency
                 value={balanceToUse ?? 0}
-                dataTestId="header-balance"
+                data-testid="header-info__account-details-modal__account-balance"
```

Comment:
Nit: This ID seems like it was intended to follow BEM naming conventions (judging by the `__` separator), but it has two "element" parts which is not compliant with BEM.

A BEM name would be something more like

```suggestion
                data-testid="header-info__account-details-modal-balance"
```

or 

```suggestion
                data-testid="confirmation-account-details-modal__account-balance"
```

Reply:
I see your point.. I wasn't, intentionally, trying to follow BEM naming conventions. I don't think that there's a convention to the `data-testid` (maybe we should have one). I think I was just splitting by "component hierarchy". Anyways, I'll update them to avoid confusion.

Reply:
Also we probably should have some guidelines/conventions on how to name `data-testid`. Otherwise we might inadvertently end up with two elements with the same `data-testid` in the dom, which can lead to unexpected issues when writing tests. Also I find myself heavily relying on the `data-testid`, which seems to be the recommended locator strategy for our e2e tests (according to the contributor docs).
However, only recommends querying by testid if it is not possible to use other queries (except querying dom structure or css classes). Their ordered recommendation for queries can be found [here](https://testing-library.com/docs/queries/about/#priority)

Reply:
done 👍